query-node/mappings/src/membership.ts

```diff
@@ -226,13 +302,9 @@ export async function members_MembershipGifted({ store, event }: EventContext &
     controllerAccount: member.controllerAccount,
     rootAccount: member.rootAccount,
     handle: member.handle,
-    metadata: new MemberMetadata({
-      ...member.metadata,
-      id: undefined,
-    }),
+    metadata: await saveMembershipMetadata(store, member),
```

Comment:
Now I see one problem that predates your PR, here in the event you are creating a metadata record using `await saveMembershipMetadata(store, member)` and then setting the event's metadata property to the record.

But we already have a member's metadata record created by the `createNewMemberFromParams` function at [line](https://github.com/Joystream/joystream/pull/4178/files#diff-9af242945476363410a798a87c9d22b687aeeea210b9b1bbc4d0697fde116b04L221). So for every member, we are creating two same metadata records in the table. What you can do is just use metadata property from the member object.

```suggestion
    metadata: member.metadata,
```

Also, this problem exists in other mappings too, such as `members_MembershipBought`, `members_MemberInvited`, can you please check those mappings too? Thanks

Reply:
Yes I think this was on purpose. As you mentioned even before this PR the membership mappings were creating one metadata record for the `Membership` and on more for the `MemberXxxEvent`. This way when `MemberProfileUpdated` is dispatched: `Membership.metadata` is mutated but all the previous `MemberXxxEvent`s are not.

I'm not sure how important it is to keep an accurate history of the membership metadata events. But because I saw this as a common pattern in the mappings, I tried to follow it here too. Although if you think this is too costly I could reset the branch to [this commit](https://github.com/Joystream/joystream/pull/4178/commits/3c8f6e2044385a7177a16bbe9b53ce8466cbf4b6) which significantly simplifies the code. I don't think it would actually affect Pioneer ATM (I'm not sure about Atlas).

Reply:
Thanks for the clarification, I didn't have the full context of why we had duplicate metadata records, your description sounds perfectly fine and I think we should have history of metadata records and try not to do any change at this stage. Everything else is fine, so approving the PR.